vmdb/app/helpers/api_helper/renderer.rb

```diff
@@ -283,19 +283,25 @@ def split_virtual_attribute(attr)
 
     def attr_accessible?(object, attr)
       return false unless object && object.respond_to?(attr)
-      is_reflection = object.class.reflections_with_virtual.keys.collect(&:to_s).include?(attr)
-      is_column = object.class.columns_hash_with_virtual.keys.include?(attr) unless is_reflection
+      is_reflection = object.class.reflection_with_virtual(attr.to_sym)
+      is_column     = object.class.columns_hash_with_virtual.key?(attr) unless is_reflection
```

Comment:
Use `virtual_field` here.  It will check the [virtual columns hash and the regular columns hash](https://github.com/manageiq/manageiq/blob/d168cdbfe0de910f61d3ae5d9347f1abb2d546cd/vmdb/lib/extensions/ar_virtual.rb#L169-L171).